src/test.rs

```diff
@@ -17,6 +18,118 @@ macro_rules! t {
     };
 }
 
+// `repo_test! will
+macro_rules! repo_test {
```

Comment:
I may be missing something, but how come a macro for this was added? Could the tests be written just against the right function?

Reply:
I asume that @mjkeeler wanted to make sure that same code should have same functionalities even if underlying repositories are different (normal repo, normal repo's worktree, bare repo, bare repo's worktree). I've finished his implementation, but I still think it is the libgit2's responsibility since git2-rs is just a binding. Should I remove this?

Reply:
I think it's good to keep some tests in this repository, but whether or not it's done with a macro is up to you. I personally prefer to avoid macros where possible.